service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -261,7 +263,37 @@ public ClientConfigProperties downstreamSenderConfig() {
     public KafkaProducerConfigProperties kafkaProducerConfig() {
         final KafkaProducerConfigProperties configProperties = new KafkaProducerConfigProperties();
         if (getAdapterName() != null) {
-            configProperties.setClientId(getAdapterName());
+            configProperties.setClientIdFallback(getAdapterName());
+        }
+        return configProperties;
+    }
+
+    /**
+     * Exposes configuration properties for a consumer accessing the Kafka cluster as a Spring bean.
+     *
+     * @return The properties.
+     */
+    @ConfigurationProperties(prefix = "hono.kafka")
+    @Bean
+    public KafkaConsumerConfigProperties kafkaConsumerConfig() {
+        final KafkaConsumerConfigProperties configProperties = new KafkaConsumerConfigProperties();
+        if (getAdapterName() != null) {
+            configProperties.setClientIdFallback(getAdapterName());
+        }
+        return configProperties;
+    }
+
+    /**
+     * Exposes configuration properties for an admin client accessing the Kafka cluster as a Spring bean.
+     *
+     * @return The properties.
+     */
+    @ConfigurationProperties(prefix = "hono.kafka")
+    @Bean
+    public KafkaAdminClientConfigProperties kafkaAdminClientConfig() {
```

Comment:
Just out of curiosity, what is the Admin client used for by the protocol adapter?

Reply:
It is to be used for creating the `hono.command_internal.${adapterInstanceId}` topic (with one partition) on startup, and to delete this topic on shutdown. 
That would at least be the preferred setup. The topic could also be auto-created but then there would be the need for cleanup when the adapter instance isn't there anymore.

Reply:
> It is to be used for creating the `hono.command_internal.${adapterInstanceId}` topic (with one partition) on startup, and to delete this topic on shutdown.

Ah, thanks. That makes sense.

> The topic could also be auto-created but then there would be the need for cleanup when the adapter instance isn't there anymore.

Yes, this would be suboptimal. Kafka admins might want to disable the broker config `auto.create.topics.enable` (and for managed Kafka instances there might be no way to enable this).